eng/NullablePolyfill.targets

```
@@ -0,0 +1,26 @@
+<Project>
+
+  <Choose>
+    <!--
+        C# 8.0's nullable annotations only work for .NET Core 3.0+ by default. However it can be made to work for other target frameworks
+        by compiling the annotations (as internal types).
+        The code block that follows sets it up so projects in this repo that target ns2.0 or netfx can compile when Nullable is configured.
+        Based on https://github.com/dotnet/runtime/blob/93b6c449d4f31ddd7d573d1d3769e681d5ebceb9/src/libraries/Directory.Build.targets#L215-L222
+
+        SuppressNullableAttributesImport to disable. Used when attributes are already present from another project because of InternalsVisibleTo.
+    -->
+    <When Condition=" ('$(Nullable)' == 'annotations' OR '$(Nullable)' == 'enable') AND
+        '$(SuppressNullableAttributesImport)' != 'true' AND
+        (('$(TargetFrameworkIdentifier)' == '.NETStandard' AND $([MSBuild]::VersionLessThanOrEquals('$(TargetFrameworkVersion)', '2.1'))) OR '$(TargetFrameworkIdentifier)' == '.NETFramework')">
```

We can really simplify this condition given that we know we never target old/end of life TFMs.

```suggestion
    <When Condition="'$(Nullable)' != '' and
                        '$(Nullable)' != 'disable' and
                        '$(SkipIncludeNullableAttributes)' != 'true' and
                        '$(MSBuildProjectExtension)' == '.csproj' and
                        '$(TargetFrameworkIdentifier)' != '.NETCoreApp'">
```

This is what currently is in `main` of dotnet/runtime:
https://github.com/dotnet/runtime/blob/4d37f40c49ed7f03d557e96323835bb87638b6ad/src/libraries/Directory.Build.targets#L171-L175

---

Yeah I lifted this straight from dotnet/aspnetcore: https://github.com/dotnet/aspnetcore/blob/542f2b9877da67e90cca6ce1b65aa9557fbb2aa8/eng/targets/CSharp.Common.targets#L112-L132

Happy to change to simpler version.